### PR TITLE
Issue/pcs 66842 keep statement timeout while switching connections

### PR DESCRIPTION
**Pull request by wyrzyk**

### CODE REVIEW

File: src/test/java/com/atlassian/db/replica/api/TestDualConnection.java [hunk 1]
```diff
@@ -1653,6 +1655,37 @@ public void shouldKeepUsingReplicaAfterSettingTimeout() throws SQLException {
...
+        final List<Statement> mainConnectionStatements = allStatements
+            .stream()
+            .filter(statement -> !replicaStatements.contains(statement))
+            .collect(Collectors.toList());
+        assertThat(connectionProvider.getProvidedConnectionTypes())
+            .containsExactly(REPLICA, MAIN);
+        assertThat(replicaStatements).hasSize(1);
+        verify(replicaStatements.get(0)).execute("set statement_timeout to 30000");
+        assertThat(mainConnectionStatements).hasSize(1);
+        verify(mainConnectionStatements.get(0)).execute("set statement_timeout to 30000");
```
Comment (afaruga-atlassian) [inline src/test/java/com/atlassian/db/replica/api/TestDualConnection.java:1686]: is there option to have different `set statement_timeout to 30000` only on MAIN ?

Comment (wyrzyk) [reply src/test/java/com/atlassian/db/replica/api/TestDualConnection.java:1686]: `set` changes configuration for a current connection. From the perspective of DualConnection's user, there's no replica/main. There's only DualConnection. The user can't use different settings for main and replica as the internal connections are transparent and there's no dedicated API to set one or another.

File: src/test/java/com/atlassian/db/replica/api/TestDualConnection.java [hunk 2]
```diff
@@ -1653,6 +1655,37 @@ public void shouldKeepUsingReplicaAfterSettingTimeout() throws SQLException {
             .containsOnly(REPLICA);
     }
 
+    @Test
```
Comment (afaruga-atlassian) [inline src/test/java/com/atlassian/db/replica/api/TestDualConnection.java:1658]: You were close in `shouldKeepUsingReplicaAfterSettingTimeout` test on covering `set statement_timeout` use case. But it wasn't tested from this perspective

File: src/main/java/com/atlassian/db/replica/internal/ConnectionParameters.java [hunk 3]
```diff
@@ -47,6 +51,24 @@ public void initialize(Connection connection) throws SQLException {
...
+            return;
+        }
+        try (Statement statement = connection.createStatement()) {
+            for (String parameterChange : runtimeParameterChanges) {
+                statement.execute(parameterChange);
+            }
+        }
+    }
+
+    public void addRuntimeParameterConfiguration(
```
Comment (afaruga-atlassian) [inline src/main/java/com/atlassian/db/replica/internal/ConnectionParameters.java:68]: for now `runtimeParameterChanges` are only `statements`, do you expect other parameters like this? if yes which?

**Review by afaruga-atlassian — APPROVED** [review]
LGTM

File: src/main/java/com/atlassian/db/replica/internal/ConnectionParameters.java (hunk 3 above)
Comment (wyrzyk) [reply src/main/java/com/atlassian/db/replica/internal/ConnectionParameters.java:68]: I think I don't understand the question. What does it mean `runtimeParameterChanges are only statements`?

File: src/test/java/com/atlassian/db/replica/api/TestDualConnection.java (hunk 1 above)
Comment (afaruga-atlassian) [reply src/test/java/com/atlassian/db/replica/api/TestDualConnection.java:1686]: Correct, I was wondering about some edge case, but it doesn't make senses as
> internal connections are transparent

👍

File: src/main/java/com/atlassian/db/replica/internal/ConnectionParameters.java (hunk 3 above)
Comment (afaruga-atlassian) [reply src/main/java/com/atlassian/db/replica/internal/ConnectionParameters.java:68]: this is NIT.
In other words, why `addRuntimeParameterConfiguration` ? 
not `addStatemets`, what do you have in mind?

Comment (wyrzyk) [reply src/main/java/com/atlassian/db/replica/internal/ConnectionParameters.java:68]: `set` sets runtime parameters for the current connection. We add the configurations (in a form of SQL commands), not statements.
